app/controllers/admin/instances/notes_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class Admin::Instances::NotesController < Admin::BaseController
+  before_action :set_instance, only: [:create]
+  before_action :set_instance_note, only: [:destroy]
+  rescue_from ActiveRecord::RecordInvalid, with: :bad_request
+
+  def create
+    authorize :instance_note, :create?
+
+    @instance_note = current_account.instance_notes.new(content: resource_params[:content], domain: @instance.domain)
+
+    if @instance_note.save
+      redirect_to admin_instance_path(@instance.domain, anchor: helpers.dom_id(@instance_note)), notice: I18n.t('admin.instances.moderation_notes.created_msg')
+    else
+      @instance_notes = @instance.moderation_notes.includes(:account).latest
+      @time_period = (6.days.ago.to_date...Time.now.utc.to_date)
+      @action_logs = Admin::ActionLogFilter.new(target_domain: @instance.domain).results.limit(5)
+
+      render 'admin/instances/show'
```

Comment:
Not sure about rendering into another controller ... I assume that's what the i-var setup above is for?

Reply:
Yeah, but we do this in a tonne of places as to not loose form state.